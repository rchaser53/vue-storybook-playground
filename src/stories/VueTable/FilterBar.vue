<template>
  <div class="filter-bar ui basic segment grid">
    <div class="ui form">
      <div class="inline field">
        <label>Search for:</label>
        <input type="text" v-model="filterText" class="three wide column" @keyup.enter="doFilter" placeholder="name, nickname, or email">
        <button class="ui primary button" @click="doFilter">Go</button>
        <button class="ui button" @click="resetFilter">Reset</button>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component'

@Component
export default class FilterBar extends Vue {
  filterText = ''

  doFilter () {
    (this as any).$events.fire('filter-set', this.filterText)
  }
  resetFilter () {
    this.filterText = '';
    (this as any).$events.fire('filter-reset')
  }
}
</script>