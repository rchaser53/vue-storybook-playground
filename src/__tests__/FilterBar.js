import Vue from 'vue';
import FilterBar from '../stories/VueTable/FilterBar.vue';
import { createRenderer } from 'vue-server-renderer';
const renderer = createRenderer();

const render = (vm) => {
  return new Promise((resolve, reject) => {
    renderer.renderToString(vm, (err, str) => {
      if (err) {
        return reject(err);
      }
      resolve(str);
    });
  });
};

test('FilterBarをrenderする', () => {
  const vm = new Vue({
  	el: document.createElement('div'),
  	components: {
  		'vue-table': FilterBar,
  	},
  	render: (h) => {
  		return h('vue-table');
  	},
  });
  expect(vm.$el).toMatchSnapshot();
});
