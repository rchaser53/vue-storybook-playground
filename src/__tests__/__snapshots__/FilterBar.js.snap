// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBarをrenderする 1`] = `
<div
  class="filter-bar ui basic segment grid"
>
  <div
    class="ui form"
  >
    <div
      class="inline field"
    >
      <label>
        Search for:
      </label>
       
      <input
        class="three wide column"
        placeholder="name, nickname, or email"
        type="text"
      />
       
      <button
        class="ui primary button"
      >
        Go
      </button>
       
      <button
        class="ui button"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
